File: LineUp/lumodule/graphic/graphic.py
from tkinter import Tk
from tkhtmlview import HTMLLabel

class Main(Exec):
    def __init__(self, global_variable, global_class):
        super(Main, self).__init__()
        self.global_variable = global_variable
        self.global_class = global_class
    
    def begin(self, *args):
        self.graphic = ["<!DOCTYPE html>", "<html>"]
    
    def end(self, *args):
        self.graphic.append("</html>")
    
    def view(self, *args):
        root = Tk()
        label = HTMLLabel(root, html="\n".join(self.graphic))
        label.pack()
        root.mainloop()
    
    def text(self, *args):
        self.graphic.append(args[0])
    
    def style(self, *args):
        self.graphic.append("style=\"")
        for i in args[0].decoded_line:
            self.graphic.append(i[1].decoded_line[0] +
                ": " + i[1].decoded_line[1]+";")
        self.graphic.append("\"")
    
    def win(self, *args):
        self.graphic.append("<"+args[0])
        for i in args[1].decoded_line:
            i[1].execute(self.global_variable, self.global_class)
        self.graphic.append(">")
        if (len(args) >= 3):
            for i in args[2].decoded_line:
                i[1].execute(self.global_variable, self.global_class)
        self.graphic.append("</"+args[0]+">")

File: LineUp/lumodule/graphic/graphic.php
<?php
class Main extends Execution
{
    public $global_variable;
    public $global_class;
    private $graphic;

    public function __construct(&$global_variable, &$global_class)
    {
        $this->global_variable = &$global_variable;
        $this->global_class = &$global_class;
    }

    public function begin($parameters)
    {
        $this->graphic = array("<!DOCTYPE html>", "<html>");
    }

    public function end($parameters)
    {
        $this->graphic[] = "</html>";
    }
    public function view($parameters)
    {
        echo implode("\n", $this->graphic);
    }
    public function text($parameters)
    {
        $this->graphic[] = $parameters[0];
    }
    public function style($parameters)
    {
        $this->graphic[] = "style=\"";
        foreach ($parameters[0]->decoded_line as $value) {
            $this->graphic[] = $value[1]->decoded_line[0].": ".$value[1]->decoded_line[1].";";
        }
        $this->graphic[] = "\"";
    }

    public function win($parameters)
    {
        $this->graphic[] = "<".$parameters[0];
        foreach ($parameters[1]->decoded_line as $value) {
            $value[1]->execute($this->global_variable, $this->global_class);
        }
        $this->graphic[] =  ">";
        if (count($parameters) >= 3) {
            foreach ($parameters[2]->decoded_line as $value) {
                $value[1]->execute($this->global_variable, $this->global_class);
            }
            $this->graphic[] = "</".$parameters[0].">";
        }
    }
}
